scripts/common_test.py

```diff
@@ -760,3 +776,26 @@ def test_swap_env_when_var_did_not_exist(self):
             self.assertIsNone(old_value)
             self.assertEqual(os.environ['DEF'], 'Easy as 123')
         self.assertNotIn('DEF', os.environ)
+
+    def test_write_stdout_safe_with_repeat_oserror(self):
+
+        raised_once = False
+
+        def write_raise_oserror(unused_fileno, bytes_to_write):
+            self.assertEqual(bytes_to_write, 'test'.encode('utf-8'))
+
+            nonlocal raised_once
+            if not raised_once:
+                raised_once = True
+                raise OSError(errno.EAGAIN, 'OS error that should be repeated')
+
+            return 4
+
+        write_swap = self.swap_with_checks(os, 'write', write_raise_oserror)
+        with write_swap:
+            common.write_stdout_safe('test')
```

Comment:
Perhaps add some kind of assertion here to verify successful completion of the operation, e.g. that 'test' actually got printed to stdout. Otherwise I can't really tell what this test is testing.

Reply:
This is done inside the `write_raise_oserror`, since we need to mock the write operation it is never written to stdout.

Reply:
But then this test would pass even if the write_raise_oserror() function wasn't called. You'd probably want calling it to append to some external list and then check that list at the end to ensure the correct thing happened, rather than putting an assertion in the function.

Reply:
No, because `swap_with_checks` makes sure that the function is always called.

Reply:
OK, let's step back a bit. What is this test actually trying to test? The test name doesn't really explain this (and should perhaps be updated) -- what is the expected result here and what would an example of a test failure be?

Reply:
Changed the name and also added the verification of args to `swap_with_checks`. This test makes sure that when write fails (with `errno.EAGAIN`) the call is repeated.

Reply:
Looks better, thanks -- just one thing, please add a line comment somewhere saying "This test makes sure that when write fails (with errno.EAGAIN) the call is repeated." The test name is slightly cryptic and that sentence of yours helped me understand it properly.

Reply:
Done, I also added an assert for `raised_once`.